eip-draft_babyjubjub.md

```diff
@@ -0,0 +1,363 @@
+
+The algorithm chosen for generating Baby Jubjub is based in the criteria defined in [[RFC7748, Appendix A.1]](https://tools.ietf.org/html/rfc7748) and can be found in [this github repository](https://github.com/barryWhiteHat/baby_jubjub). Essentially, the algorithm takes a prime number `p = 1 mod 4` and returns the lowest `A>0` such that `A-2` is a multiple of 4 and such that the set of solutions in `F_p` of `y^2 = x^3 + Ax^2 + x` defines a Montgomery curve with cofactor 8. 
+
+Baby Jubjub was generated by running the algorithm with the prime
+
+`r =  21888242871839275222246405745257275088548364400416034343698204186575808495617`, 
+
+which is the order of alt_bn128, the curve used to verfiy zk-SNARK proofs in Ethereum. The output of the algorithm was `A=168698`. Afterwards, the corresponding Montgomery curve was transformed into twisted Edwards form. Using SAGE libraries for curves, the order `n` of the curve and its factorization `n = 8*l` was calculated.
+
+- **Choice of generator** : the generator point `G` is the point of order `n` with smallest positive `x`-coordinate in `F_r`. 
+- **Choice of base point**: the base point `B` is chosen to be `B = 8*G`, which has order `l`. 
+
+**Security considerations**
+
+It is crucial that Baby Jubjub be safe against well-known attacks. To that end, we decided that the curve should pass [SafeCurves](https://safecurves.cr.yp.to/) security tests, as they are known for gathering the best known attacks against elliptic curves. Supporting evidence that Baby Jubjub satisfies the SafeCurves criteria can be found [here](https://github.com/barryWhiteHat/baby_jubjub).
+
+
+# Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+
+Baby Jubjub is a twisted Edwards elliptic curve birational to different curves. So far, the curve has mainly been used in its original form, in Montomgery form. and in another (different representation) twisted Edwards form -- which we call the reduced twisted Edwards form.
+
+Below are the three representations and the birational maps that make it possible to map points from one form of the curve to another. In all cases, the generator and base points are written in the form **`(x,y)`.**
+
+## Forms of the curve
```

Comment:
```suggestion
### Forms of the Curve
```

Reply:
Why is this in the Backward Compatability section?  Should probably be in the specification section, or maybe some appendix.

Reply:
We decided to include these two other forms of the curve in the Backward Compatibility section because other projects were already using them. So, although we chose to standardize the curve as we presented it, we thought it was important to add a section reflecting the other valid forms that were already being used and explaining how was possible to get from those to the standard form and viceversa. (We did not include any variation of the curve, only the two that were already out there). I hope this clarifies it!